jcommune-view/jcommune-web-view/src/test/java/org/jtalks/jcommune/test/SignInTest.groovy

```diff
@@ -0,0 +1,112 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+package org.jtalks.jcommune.test
+
+import org.jtalks.jcommune.model.utils.Groups
+import org.jtalks.jcommune.test.utils.Users
+import org.jtalks.jcommune.test.utils.exceptions.WrongResponseException
+import org.jtalks.jcommune.test.utils.model.User
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.test.context.ContextConfiguration
+import org.springframework.test.context.transaction.TransactionConfiguration
+import org.springframework.test.context.web.WebAppConfiguration
+import org.springframework.test.web.servlet.setup.MockMvcBuilders
+import org.springframework.transaction.annotation.Transactional
+import org.springframework.web.context.WebApplicationContext
+import spock.lang.Specification
+
+import javax.annotation.Resource
+import javax.servlet.Filter
+
+import static org.apache.commons.lang.RandomStringUtils.randomAlphabetic
+
+/**
+ * @author Mikhail Stryzhonok
+ */
+@WebAppConfiguration
+@ContextConfiguration(locations = [
+        'classpath:/org/jtalks/jcommune/model/entity/applicationContext-dao.xml',
+        'classpath:/org/jtalks/jcommune/model/entity/applicationContext-properties.xml',
+        'classpath:/org/jtalks/jcommune/service/applicationContext-service.xml',
+        'classpath:/org/jtalks/jcommune/service/security-service-context.xml',
+        'classpath:/org/jtalks/jcommune/service/email-context.xml',
+        'classpath:/org/jtalks/jcommune/web/applicationContext-controller.xml',
+        'classpath:security-context.xml',
+        'classpath:spring-dispatcher-servlet.xml',
+        'classpath:/org/jtalks/jcommune/web/view/test-configuration.xml'
+])
+@TransactionConfiguration(transactionManager = 'transactionManager', defaultRollback = true)
+@Transactional
+abstract class SignInTest extends Specification {
+
+    @Autowired
+    private WebApplicationContext ctx
+
+    protected Users users
+    @Autowired
+    private Groups groups
+
+    @Resource(name = 'testFilters')
+    List<Filter> filters
+
+    def setup() {
+        users.mockMvc = MockMvcBuilders.webAppContextSetup(ctx)
+                .addFilters(filters.toArray(new Filter[filters.size()])).build()
+        groups.create()
+    }
+
+    def 'Sign in without activation registration should fail'() {
+        given: 'User is registered and not activated'
+            def user = new User()
+            users.singUp(user)
+        when: 'User tries to log in'
+            users.signIn(user)
+        then: 'Wrong response exception is thrown'
+            thrown(WrongResponseException)
```

Comment:
It will be difficult because we don't pass any errors to response. For signing in via jsp page we pass query parameter which indicates error and analyze this parameter inside jsp. For signing in via pop up we also not pass error info into response. And for signing in via pop up all errors except incorrect username/password are shown in separate dialog.